portal/src/main/webapp/config_service.jsp

```diff
@@ -88,8 +63,106 @@ studies += "}";
-  
     
-  
-});
+   
+};
+--%>
+
+// an empty string means the property is listed but not assigned to in portal.props 
+// a null value means it was entirely missing 
 
+<%= request.getParameter("callback") %>({
+
+    <%
+    
+        // Set API root variable for cbioportal-frontend repo
+        String url = request.getRequestURL().toString();
+        String baseURL = url.substring(0, url.length() - request.getRequestURI().length()) + request.getContextPath();
+        baseURL = baseURL.replace("https://", "").replace("http://", "");
+    
+        String[] propNameArray = {
+            "app.version",
+            "oncoprint.custom_driver_annotation.binary.menu_label",
+            "disabled_tabs",            
+            "civic.url",
+            "oncoprint.custom_driver_annotation.default",
+            "oncoprint.oncokb_hotspots.default",
+            "genomenexus.url",
+            "google_analytics_profile_id",
+            "oncoprint.hide_vus.default",
+            "mycancergenome.show",
+            "oncokb.public_api.url",
+            "digitalslidearchive.iframe.url",
+            "digitalslidearchive.meta.url",
+            "mdacc.heatmap.meta.url",
+            "mdacc.heatmap.patient.url",
+            "priority_studies",
+            "show.hotspot",
+            "show.oncokb",
+            "show.civic",
+            "skin.documentation.about",
+            "skin.documentation.baseurl",
+            "skin.blurb",
+            "skin.custom_header_tabs",
+            "skin.data_sets_footer",
+            "skin.data_sets_header",
+            "skin.documentation.markdown",
+            "skin.email_contact",
+            "skin.example_study_queries",            
+            "skin.examples_right_column_html",
+            "skin.documentation.faq",
+            "skin.footer",
+            "skin.login.contact_html",
+            "skin.login.saml.registration_html",
+            "skin.documentation.news",
+            "skin.documentation.oql",
+            "skin.query.max_tree_depth",
+            "skin.right_logo",
+            "skin.right_nav.show_data_sets",
+            "skin.right_nav.show_examples",
+            "skin.right_nav.show_testimonials",
+            "skin.right_nav.whats_new_blurb",
+            "skin.show_about_tab",
+            "skin.show_data_tab",
+            "skin.show_faqs_tab",
+            "skin.show_news_tab",
+            "skin.show_r_matlab_tab",
+            "skin.show_tools_tab",
+            "skin.show_tutorials_tab",
+            "skin.show_web_api_tab",
+            "skin.title",
+            "skin.authorization_message",
+            "mdacc.heatmap.study.meta.url",
+            "mdacc.heatmap.study.url",
+            "oncoprint.custom_driver_annotation.tiers.menu_label"
+        }; 
+    
+        // for each above, add json prop and lookup value in portal.properties
+        for (int i = 0; i < propNameArray.length; i++){
```

Comment:
i'm sure there is some better way to serialize all this stuff but this works pretty well.  actually, there i still one problem which is that boolean values are delivered as strings.  I correct this on frontend but we should figure that out here.

Reply:
ditto numeric values